lib/pbench/server/templates.py

```diff
@@ -609,29 +697,33 @@ def dump_templates(self):
         sys.stdout.flush()
 
     def update_templates(self, es, target_name=None):
-        """Push the various Elasticsearch index templates required by pbench.
+        """
+        Register with Elasticsearch the set of index templates used by the
+        Pbench server.
+
+        Args
+            es:             Elasticsearch object to connect to server
+            target_name:    Optional template name to update just one template
+
+        Raises
+            TemplateError   Problem updating the template to server
         """
         if target_name is not None:
             idxname = TemplateFile.index_patterns[target_name]["idxname"]
         else:
             idxname = None
         template_names = {t.template_name: t for t in self.templates.values()}
-        names = list(template_names.keys())
-        names.sort()
         successes = retries = 0
         beg = end = None
-        for name in names:
+        for name in sorted(template_names.keys()):
             template = template_names[name]
             if idxname is not None and not target_name == template.idxname:
```

Comment:
Evolution moves through a series of forms that work. The shape of the `self.templates` dict and the `TemplateFile` object changed a number of times, and in many of the mutations I struggled to make all of these methods retain the same behavior. This indirection got left behind because some form of it was necessary at one point and when it mutated back into this useless form it didn't break anything and I didn't notice.